api/types/appserver.go

```diff
@@ -308,3 +309,64 @@ func (s AppServers) Less(i, j int) bool {
 
 // Swap swaps two app servers.
 func (s AppServers) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+
+// SortByCustom custom sorts by given sort criteria.
+func (s AppServers) SortByCustom(sortBy *SortBy) error {
+	if sortBy == nil {
```

Comment:
`SortBy` is just a bool and two pointers, and a non-nil `SortBy` with `Field: ""` is invalid anyway.
```suggestion
func (s AppServers) SortByCustom(sortBy SortBy) error {
	if sortBy.Field == "" {
```

Reply:
Matching this, the `SortBy` field in `ListResourcesRequest` in [api/client/proto/authservice.proto](https://github.com/gravitational/teleport/pull/10243/files#diff-5abc0b0ed5cf2b1a7c43d4e5bdb7f45b215c1c7896c2fcc7ebec85402ffe59b1R1404) could be made non-nullable.